### ISSUE TITLE
Regression in 0.27 user agent handling breaks Construct 2 content

### ISSUE THREAD

**Issue by AshleyScirra** [nwjs/nw.js#6321]
NWJS Version : 0.27.0
Operating System : Windows 10

### Expected behavior

With NW.js 0.27, if package.json specifies a `user-agent`, it is not applied to navigation requests.  It is correctly applied to all subresource requests, but not the main HTML document request. NW.js 0.26 was not affected. This has a serious impact for Construct 2, since it relies on the correct user agent (which specifies NW.js) in order to enable NW.js features. (E.g. this report: https://www.scirra.com/forum/nw-js-plugin-broken-in-preview-v0-27-0_t198544)

### Actual behavior

With NW.js 0.27, the navigation request reverts to using the default user agent ("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.84 Safari/537.36"), ignoring the user agent set in package.json.

### How to reproduce
Use the following package.json. Note the term "MySuperUserAgent" in the UA string.
```
{
	"main": "https://nwjs.io/",
	"name": "UA test",
	"user-agent": "Mozilla/5.0 (%osinfo) AppleWebKit/%webkit_ver (KHTML, like Gecko, Chrome, Safari) MySuperUserAgent/1.0"
}
```
Launch NW.js and open dev tools to the network tab. You may need to reload the page with cache disabled to see all requests. Observe HTTP request headers for:
1) the first nwjs.io request (UA is: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.84 Safari/537.36)
2) a later subresource request, e.g. for index.js (UA is: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (@f340e0691c0c44d924c05db57ed0c8324064dcda) (KHTML, like Gecko, Chrome, Safari) MySuperUserAgent/1.0)

Note that "MySuperUserAgent" does not occur in 1) but does in 2).

**Comment by rogerwang** [comment]
This is just fixed in git and will be available in the next nightly build.

**Comment by AshleyScirra** [comment]
Thanks for the quick fix. Are you planning on an NW.js 0.27.2 release?

**Comment by rogerwang** [comment]
Is the last nightly works well for you?
https://dl.nwjs.io/live-build/12-12-2017/nw27-c646a38-bed4837-5926d65-eb0dc2e/v0.27.2/

**Comment by AshleyScirra** [comment]
Yes, that build you linked to works OK with Construct 2.

**Comment by rogerwang** [comment]
That build is in release quality and it contains only the fix based on 0.27.1.

We plan to release 0.27.2 after updating Node.js to latest 9.3.0.

**Comment by AshleyScirra** [comment]
Great stuff, thanks.

**Comment by AshleyScirra** [comment]
We released the 0.27.2 update today and everything's working again. Thanks for the quick turnaround!
